cpp/daal/src/algorithms/dbscan/dbscan_distributed_input_types.cpp

```diff
@@ -129,21 +129,23 @@ services::Status DistributedInput<step2Local>::check(const daal::algorithms::Par
     const size_t nBlocks = dcPartialData->size();
     DAAL_CHECK_EX(nBlocks > 0, ErrorIncorrectNumberOfInputNumericTables, ArgumentName, partialDataStr());
 
-    size_t nFeatures = 0;
-    for (size_t i = 0; i < nBlocks; i++)
+    if (NumericTable::cast((*dcPartialData)[0])->getNumberOfRows() == 0)
```

Comment:
First, this check is incorrect. It should be `getNumberOfRows() != 0` to make sense.
Second, it is reasonable to put this check inside loop and just skip tables with `0` rows.